src/sid/shared.py

```diff
@@ -45,13 +46,17 @@ def factorize_assortative_variables(states, assort_by):
           correspond the values of assortative variables to form the group.
 
     """
-    if assort_by:
+    if is_recurrent:
+        assert isinstance(assort_by, list)
+        assert len(assort_by) == 1
```

Comment:
Can we get rid of the asserts? In case both assert conditions are False, can we use the elif case?

Reply:
I think we can just completely remove them. The condition is also checked elsewhere.